libs/ui-toolkit/src/components/drawer/drawer.tsx

```diff
@@ -15,6 +17,19 @@ interface DrawerProps {
   trigger?: ReactNode;
 }
 
+export const DrawerContext = createContext<true | undefined>(undefined);
+
+type DrawerStore = {
+  drawerOpen: boolean;
+  setDrawerOpen: (isOpen: boolean) => void;
+};
+export const useDrawer = create<DrawerStore>((set) => ({
+  drawerOpen: false,
+  setDrawerOpen: (isOpen) => {
+    set({ drawerOpen: isOpen });
+  },
+}));
+
 export function Drawer({
```

Comment:
**nitpick (blocking)**: Scolling can get weird on mobile - although here I am testing with Firefox, so if the issue doesn't show up on iOS safari/Chrome it's less blocking.

https://user-images.githubusercontent.com/6678/223148594-0df312c3-3c07-4b38-af67-aad58d1a4ed2.mov